scripts/30_prepare_disk.sh

```diff
@@ -208,6 +208,51 @@ partition_resize()
     sfdisk --quiet "${TARGET_STORAGE_DEVICE}" < "${SYSTEM_UPDATE_DIR}/${PARTITION_TABLE_FILE}"
 }
 
+backup_data()
+{
+    temp_mount_dir=$(mktemp -d)
+
+    sfdisk --quiet --dump "${TARGET_STORAGE_DEVICE}" | \
+    while IFS="${IFS}:=," read -r disk_device _ disk_start _ disk_size _; do
+        if grep -q "${disk_device}" /proc/mounts; then
+            umount "${disk_device}"
+        fi
+        backup_file="/tmp/backup${disk_device}.tar.gz"
+        if mount "${disk_device}" "${temp_mount_dir}"; then
+            mkdir -p "$(dirname "${backup_location}")"
+            if ! tar -czf "${backup_location}" -C "${temp_mount_dir}" .; then
+                rm "${backup_location}"
```

Comment:
I think you should redirect the output to /dev/null not to pollute the output data and echo that the backup for disk device partition x was successful. 
Same goes for restore.